arcgis-ios-sdk-samples/Edit data/Edit and sync features/EditAndSyncFeaturesViewController.swift

```diff
@@ -86,21 +80,14 @@ class EditAndSyncFeaturesViewController: UIViewController {
                 self.presentAlert(error: error)
             } else {
                 let featureServiceInfo = self.geodatabaseSyncTask.featureServiceInfo!
-                let map = self.mapView.map!
-                for index in featureServiceInfo.layerInfos.indices {
+                for index in featureServiceInfo.layerInfos.indices.reversed() {
                     // For each layer in the serice, add a layer to the map.
                     let layerURL = self.featureServiceURL.appendingPathComponent(String(index))
```

Comment:
Looking at this now, I'm wondering if the layer ID should be used instead of an arbitrary index?

```swift
let featureLayers = featureServiceInfo.layerInfos.compactMap { (layerInfo) -> AGSFeatureLayer? in
    let layerID = layerInfo.id
    guard layerID >= 0 else { return nil }
    let layerURL = self.featureServiceURL.appendingPathComponent(String(layerID))
    let featureTable = AGSServiceFeatureTable(url: layerURL)
    return AGSFeatureLayer(featureTable: featureTable)
}
self.mapView.map?.operationalLayers.addObjects(from: featureLayers)
```

Reply:
Applied!